code/modules/mob/living/simple_animal/bot/griefsky.dm

```diff
@@ -185,29 +185,44 @@
 /mob/living/simple_animal/bot/secbot/griefsky/explode()
 	walk_to(src,0)
 	visible_message("<span class='boldannounceic'>[src] lets out a huge cough as it blows apart!</span>")
-	var/turf/Tsec = get_turf(src)
-	new /obj/item/assembly/prox_sensor(Tsec)
-	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(Tsec)
+	var/turf/explode_turf = get_turf(src)
+	new /obj/item/assembly/prox_sensor(explode_turf)
+	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(explode_turf)
 	Sa.build_step = 1
 	Sa.overlays += "hs_hole"
 	Sa.created_name = name
 	if(prob(50))
-		new /obj/item/robot_parts/r_arm(Tsec)
+		new /obj/item/robot_parts/r_arm(explode_turf)
 	if(prob(50)) //most of the time weapon will be destroyed
-		new weapon(Tsec)
+		new weapon(explode_turf)
 	if(prob(25))
-		new weapon(Tsec)
+		new weapon(explode_turf)
 	if(prob(10))
-		new weapon(Tsec)
+		new weapon(explode_turf)
 	if(prob(5))
-		new weapon(Tsec)
+		new weapon(explode_turf)
 	do_sparks(3, 1, src)
 	new /obj/effect/decal/cleanable/blood/oil(loc)
-	..()
-
-//this section is blocking attack
+	qdel(src)
 
-/mob/living/simple_animal/bot/secbot/griefsky/bullet_act(obj/item/projectile/P) //so uncivilized
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/secbot/griefsky/disassemble()
+	var/turf/disassemble_turf = get_turf(src)
+	new /obj/item/assembly/prox_sensor(disassemble_turf)
+	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(disassemble_turf)
+	Sa.build_step = 1
+	Sa.overlays += "hs_hole"
+	Sa.created_name = name
+	drop_part(robot_arm, disassemble_turf)
+	if(weapon == /obj/item/melee/energy/sword/saber)
```

Comment:
Is weapon a type or an item here? You're comparing it to a type directly and I'm not sure that'll work

Reply:
I am not too sure what to tell you apart that It does work correctly, it checks for the type of sword that was used during construction and stored in the ``weapon`` var of the griefsky. I am not sure how to make it check for an item instead or why we would want to